django_project/users/utils.py

```diff
@@ -0,0 +1,16 @@
+from django.shortcuts import redirect
+
+
+def handle_no_permission(request, slug):
+    """
+    Handles the case when the user is not logged in and tries to access the view.
+    Redirects the user to the login page, and after successful login, redirects
+    back to the original page with the appropriate slug to submit the comment.
+    """
+    login_url = f"/login/?next=/post/{slug}/" + "#comments-section"
+
+    if not request.user.is_authenticated:
+        return redirect(login_url)
+
+    # If the user is logged in, redirect to the original page to submit the comment
+    return redirect("post-detail", slug=slug)
```

Comment:
In what scenario would this function be called if the user was logged in? I believe all logged-in users can comment.

I am just not seeing the purpose of this last return statement. Maybe it needs to return a 500 or something>

If we decide to keep this, please improve the comment to something like this:

> "If the user is logged in, but doesn't have permission to comment, redirect to the original post"

Reply:
I observed there's no need for this, so I took it off